statsmodels/tsa/statespace/dynamic_regression.py

```diff
@@ -0,0 +1,321 @@
+        d = np.zeros(0)
+        if mod["irregular"]:
+            d = np.r_[d,np.ones(1)]
+        if mod["local_level"]:
+            d = np.r_[d,np.ones(1)]
+        if mod["trend"]:
+            d = np.r_[d,np.zeros(1)]
+        return d
+
+    @property
+    def initial_transition(self):
+        """Initial transition matrix"""
+        transition = np.zeros((self.k_states,self.k_states))
+        start = 0
+        for mod in self.exog_models:
+            k_components = sum([bool(component) for component in mod.values()])
+            end = start + k_components
+            if mod['local_level'] and not mod['trend']:
+                t = np.ones((1,1))
+            else:
+                t = np.array([[1,1],[0,1]])
+            if mod['irregular']:
+                t = block_diag(np.zeros((1,1)),t)
+            transition[start:end,start:end] = t
+            start = end
+        return transition
+
+    
+    @property
+    def initial_selection(self):
+        """Initial selection matrix"""
+        selection = np.zeros((self.k_states,self.k_state_cov))
+        start = 0
+        for mod in self.exog_models:
+            k_components = 0
+            if mod['local_level'] != "deterministic":
+                s = np.ones((1,1))
+                k_components += 1
+            if mod['trend'] & (mod['trend']!="deterministic"):
+                s = block_diag(s,np.ones((1,1)))
+                k_components += 1
+            if mod['irregular']:
+                s = block_diag(np.zeros((1,1)),s)
+                k_components += 1
+            end = start + k_components
+            selection[start:end,start:end] = s
+        return selection
+
+    def update(self, params, transformed=True, **kwargs):
+        """
+        Update the parameters of the model
+
+        Updates the representation matrices to fill in the new parameter
+        values.
+
+        Parameters
+        ----------
+        params : array_like
+            Array of new parameters.
+        transformed : boolean, optional
+            Whether or not `params` is already transformed. If set to False,
+            `transform_params` is called. Default is True..
+
+        Returns
+        -------
+        params : array_like
+            Array of parameters.
+        """
+        obs_cov = params[0]
+        state_cov_params = params[1:]
+        self.ssm.obs_cov[0,0,0]= obs_cov
+        self.ssm.state_cov[:,:,0][np.diag_indices(
+            self.k_state_cov)] = state_cov_params
+        
+
+    def transform_params(self, unconstrained):
+        """
+        Transform unconstrained parameters used by the optimizer to constrained
+        parameters used in likelihood evaluation.
+
+        Used primarily to enforce stationarity of the autoregressive lag
+        polynomial, invertibility of the moving average lag polynomial, and
+        positive variance parameters.
+
+        Parameters
+        ----------
+        unconstrained : array_like
```

Comment:
array_like is for inputs that could be various classes, e.g. ndarray, Series, DataFrame.  Should be ndarray or (I don't like) array (since array is a python type, and confuses some introspection)

Reply:
In this case, I think array_like is okay since we might actually have `pd.Series` (e.g. if you were to pass in the `params` from a previously estimated results object).

Reply:
I think I was looking at the output, which is always a ndarray. I take your point on the input.

Reply:
Good point - I should change that in the base class too.